feat: DIA-1925: [FE] “View sample raw json” ability in the Data Import page

nick-skriabin opened the pull request:
Adding import data preview based on label contig.

<img width="2544" alt="image" src="https://github.com/user-attachments/assets/c163fc62-4b29-4755-943e-80325fafcfc9" />


<!--

This description MUST be filled out for a PR to receive a review. Its primary purposes are:

 - to enable your reviewer to review your code easily, and
 - to convince your reviewer that your code works as intended.

Some pointers to think about when filling out your PR description:
 - Reason for change: Description of problem and solution
 - Screenshots: All visible changes should include screenshots.
 - Rollout strategy: How will this code be rolled out? Feature flags / env var / other
 - Testing: Description of how this is being verified
 - Risks: Are there any known risks associated with this change, eg to security or performance?
 - Reviewer notes: Any info to help reviewers approve the PR
 - General notes: Any info to help onlookers understand the code, or callouts to significant portions.

You may use AI tools such as Copilot Actions to assist with writing your PR description (see https://docs.github.com/en/copilot/using-github-copilot/using-github-copilot-for-pull-requests/creating-a-pull-request-summary-with-github-copilot); however, an AI summary isn't enough by itself. You'll need to provide your reviewer with strong evidence that your code works as intended, which requires actually running the code and showing that it works.

-->

nick-skriabin committed:
Code block card

nick-skriabin committed:
Detect project configuration

nick-skriabin committed:
Merge branch 'develop' into fb-dia-1925/sample-help

nick-skriabin committed:
Load sample data

nick-skriabin committed:
Merge branch 'develop' into fb-dia-1925/sample-help

nick-skriabin committed:
Use shared project atom to deliver config

yyassi-heartex commented on web/apps/labelstudio/src/pages/CreateProject/Import/Import.jsx:
```diff
@@ -374,42 +402,66 @@ export const ImportPage = ({
+                  Set up your{" "}
+                  <a
+                    href="#"
+                    onClick={(e) => {
+                      e.preventDefault();
+                      e.stopPropagation();
+                      openLabelingConfig?.();
+                    }}
```
slight usage improvement - create a `useCallback` with `openLabelingConfig` as a dependency

yyassi-heartex commented on web/libs/ui/src/lib/simple-card/index.tsx:
```diff
@@ -15,13 +15,17 @@ export function SimpleCard({
 >) {
   const className = cn("bg-transparent", cls);
+  const hasHeaderContent = Boolean(title || description);
+  const contentClass = cn("p-4", { "pt-0": hasHeaderContent });
```
wouldn't this add the prefix? i thought p-4 is tailwind css class. should `clsx` be used here instead?

nick-skriabin replied:
`cn` in this case is shadcn's utility that combines clsx and tailwind-merge

yyassi-heartex commented on web/apps/labelstudio/src/pages/CreateProject/Import/Import.jsx:
```diff
@@ -374,42 +402,66 @@ export const ImportPage = ({
+                    <dt>Common Formats</dt>
+                    <dd>{supportedExtensions.common.join(", ")}</dd>
+                  </dl>
+                  <b>
+                    * – Support depends on the browser
+                    <br />* – Direct media uploads have{" "}
+                    <a href="https://labelstud.io/guide/tasks.html#Import-data-from-the-Label-Studio-UI">limitations</a>{" "}
+                    and we strongly recommend using{" "}
+                    <a href="https://labelstud.io/guide/storage.html" target="_blank" rel="noreferrer">
+                      Cloud Storage
+                    </a>{" "}
+                    instead
+                  </b>
```
we need to hide these documentation links for white labeled environments

nick-skriabin replied:
it's OSS and the code is not shared with LSE

yyassi-heartex replied:
ahh ok, we have another import component in LSE 👍

nick-skriabin committed:
Cleanup

AndrejOros commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/13941879776)

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-dia-1925/sample-help'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13941879776

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13941879776

nick-skriabin committed:
Use correct feature flag

nick-skriabin committed:
Fix biome

matt-bernstein committed:
set sample annotation's user id to be the current user

AndrejOros commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/13987400617)

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-dia-1925/sample-help'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13987400617

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13987400617
